pennylane/operation.py

```diff
@@ -1094,6 +1094,28 @@ def queue(self, context=qml.QueuingContext):
         context.append(self)
         return self  # so pre-constructed Observable instances can be queued and returned in a single statement
 
+    def __add__(self, other):
+        r"""The addition operation between operators."""
+        return qml.ops.math.Sum(self, other)
+
+    def __sub__(self, other):
+        r"""The subtraction operation between operators."""
+        return self.__add__(other.__mul__(-1))
+
+    def __mul__(self, a):
+        r"""The scalar multiplication for operators."""
+        return qml.ops.math.ScalarProd(a, self)
+
+    __rmul__ = __mul__
+
+    def __matmul__(self, other):
+        r"""The multiplication between operators."""
+        return qml.ops.math.Prod(self, other)
```

Comment:
I will explain it in the ADR in more detail, sneak peek:

> It is not obvious why a tensor product is included in a normal operator product ``@``, so let's have a look at this. Consider the example of the tensor product of two single-qubit gates :math:`O_1 \otimes O_2` acting on different wires. This is short notation for :math:`(O_1 \otimes 1)@(1 \otimes O_2)`. We can always turn a tensor product into a _product of tensor products_ with identities this way, because tensor products are used to connect objects from different Hilbert spaces (i.e. operators acting on different wires). PennyLane operators already represent identities implicitly, since they refer to wires: :math:`O_1 \otimes 1` is represented by ``O(wires=[0])`` if we assume a global wire order of ``[0, 1]``. This is why :math:`O_1 \otimes O_2` is well defined as ``O1(wires=[0]) @ O2(wires=[1])``, and a unique matrix can be specified if we define ``wire_order = [0, 1]``. These considerations tell us that a separate treatment of product and tensor product is not necessary internally.